src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -71,9 +71,17 @@ public void SetActiveDevice(string? deviceSerialNumber)
         {
             _processManager.DeviceSerial = deviceSerialNumber ?? string.Empty;
 
+            _api = GetAPIVersion();
+
             _log.LogInformation($"Active Android device set to serial '{deviceSerialNumber}'");
         }
 
+        private int GetAPIVersion()
+        {
+            var output = RunAdbCommand("shell getprop ro.build.version.sdk");

```

Comment:
Should we also null `_api` when we call `SetActiveDevice` again?

Reply:
Sorry, I don't understand why null - to rewrite old value? Or maybe you mean update `_api`?

Reply:
I guess when we switch between devices, the API can change so I would null the `_api` field so that we know we need to fetch it again

Reply:
but I updated `_api` inside `SetActiveDevice` - it should help in that case

Reply:
Ah, sorry, my bad. Then we're all good